client/src/components/Carousel.jsx

```diff
@@ -28,17 +27,33 @@ const Carousel = ({ productId, carouselType }) => {
   });
 
   useEffect(() => {
-    // if carouselType is related
-    axios.get(`/products/${productId}/related`)
-      .then(response => {
-        setProducts(response.data);
-        // Have to pass in window width as first render has not occurred and number of products since setProducts is async
-        // qqq Add an OR clause for when container has maxwidth
-        scrollCarousel(0, window.innerWidth, response.data.length);
-      })
-      .catch(err => err);
-    // if carouselType is outfit
-      // get all products saved in outfit and setProducts
+    // this is where each card's information object will be created. Per product:
+    // 1. Product information
+    // 2. Review data
+    // 3. Default style picture URL
+    const createdProductCards = [];
+    related.forEach(rel => {
+      let totalRating = 0;
+      let numberOfReviews = 0;
+      for (const ratingNumber in reviews.ratings) {
```

Comment:
This calculation for the star rating and rating is present in other components including Ratings & Reviews. We could make it modular since it will be reused several times. It's not critical since the scope is more like 2-4 reuses but something to consider.

Reply:
perf